lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -476,19 +474,21 @@ def getvm(self, name=None, uuid=None, folder=None):
         if uuid:
             vm = self.content.searchIndex.FindByUuid(uuid=uuid, vmSearch=True)
         elif folder:
-            # Build the absolute folder path to pass into the search method
-            if not self.params['folder'].startswith('/'):
-                self.module.fail_json(msg="Folder %(folder)s needs to be an absolute path, starting with '/'." % self.params)
-            searchpath = '%(datacenter)s%(folder)s' % self.params
+            # searchpaths do not need to be absolute
+            searchpath = self.params['folder']
```

Comment:
I have tested, and found that **_FindByInventoryPath_** does need the full path to find a folder in both my scenarios where VMs are nested like so.
/TestDCFolder/TestDC/TestCluster/TestFolder/
and
/TestDC/TestCluster/TestFolder/

In both cases, I needed to change searchpath to match the full path to the DC, then /vm/ and the actual folder.  So in the above instances it was.
/TestDCFolder/TestDC/vm/TestFolder/
and
/TestDC/vm/TestFolder/